### DISCUSSION TITLE
[Issue] toArray() Data class method being called when returning a Paginated Collection but not with a single Data resource

### DISCUSSION THREAD

**Discussion by didac-adria** [spatie/laravel-data#175] [General]
Hello!

@albert-mestres and I have been playing recently quite a lot with this package and we love it. We actually updated to the 2.0 version.

Now, we have two endpoints in which one is returning a PaginatedDataCollection and another which is returning directly a single Data object. For the paginated one, the toArray() method in the Data object is being called, while when we return directly the Data object from the controller, it is not being called.

We expected that the toArray() method was called always when a Data object is returned in a response. In any case, we think that could be an issue because of the different behaviours mentioned.

But on top of that, we also have this issue: https://github.com/spatie/laravel-data/discussions/177
So to solve this current issue (#175), we had to call: `->toArray()`everywhere. But it turns out that, if we combine both `->wrap('data')->toArray()` the ->wrap() method (solution of issue #177) is not working.

**Comment by didac-adria** [comment]
Let me add a second indication that this may be a bug. When we have a CollectionOf Data objects, when you call the ->toArray() method on the collection Instance, the ->toArray() methods on each item are not called.

**Comment by rubenvanassche** [comment]
Please do not depend on overwriting `toArray`, the package uses it indeed for `PaginatedDataCollection` since it's the only way to create an array with meta, links, etc from Laravel PaginatedCollections. 

Internally we try to use the transform method on a data object as much as possible because it has some extended information about the kind of transformation required.

If you want to overwrite behavior, then please overwrite the `transform` method, it will be called by `toResponse`, `toJson`, `toArray` and more.
